comms/core/src/noise/mod.rs

```diff
@@ -33,3 +33,9 @@ pub use error::NoiseError;
 
 mod socket;
 pub use socket::NoiseSocket;
+use tari_utilities::{hidden_type, safe_array::SafeArray, Hidden};
+use zeroize::Zeroize;
+
+pub(crate) const NOISE_KEY_LEN: usize = 32;
+
+hidden_type!(CommsNoiseKey, SafeArray<u8, NOISE_KEY_LEN>);
```

Comment:
Do you really need to use both a SafeArray and Hidden? 

`CommsNoiseKey(Hidden(Box<SafeArray(Vec<u8>>))`  :fearful: 

Surely `hidden_type!(CommsNoiseKey, [u8; NOISE_KEY_LEN])` is sufficient as we dont need constant time comparison.

Reply:
Definitely not ! Let me refactor it to use a plain array

Reply:
Definitely not ! Let me refactor it to use a plain array, instead